src/main/java/duke/command/AddItemCommand.java

```diff
@@ -0,0 +1,92 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.Deadline;
+import duke.task.DukeTask;
+import duke.task.Event;
+import duke.task.ToDo;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Encapsulates adding items to a TaskList object.
+ *
+ * @author Kartikeya
+ */
+public class AddItemCommand implements Command {
+    private final String[] input;
+
+    /**
+     * Constructor for an AddItemCommand object.
+     *
+     * @param input split strings input by user
+     */
+    public AddItemCommand(String[] input) {
+        this.input = input;
+    }
+
+    /**
+     * {@inheritDoc}
+     * Adds given input item to itemList, and shows the resulting message to the user.
+     */
+    @Override
+    public void execute(TaskList itemList, Ui ui, Storage storage) throws DukeException {
+        ui.showToUser(executeAddItem(input, itemList));
+    }
+
+    /**
+     * Executes item addition commands, throwing a duke.exception.DukeException on incorrect
+     * inputs.
+     *
+     * @param newItem Parsed item addition command
+     * @return string signifying result of adding newItem to items
+     * @throws DukeException Indicates incorrect inputs
```

Comment:
Maybe you could consider rephrasing it towards @throws DukeException If (something occurs)?